lib/publiccloud/ssh_interactive.pm

```diff
@@ -144,6 +144,7 @@ sub prepare_ssh_tunnel {
 
     # Permit root passwordless login over SSH
     $instance->ssh_assert_script_run('sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
+    $instance->ssh_assert_script_run('sudo sed -i "s/#AllowTcpForwarding yes/AllowTcpForwarding yes/g" /etc/ssh/sshd_config');
```

Comment:
I would recommend to match for lines containing anything `AllowTcpForwarding ` and replace the whole line, e.g.

```suggestion
    $instance->ssh_assert_script_run('sudo sed -i "/AllowTcpForwarding /c\AllowTcpForwarding yes" /etc/ssh/sshd_config');
```

Reply:
I agree. Shouldn't it be `'s/.*AllowTcpForwarding.*/AllowTcpForwarding yes/'` ?

Reply:
Agreed to move forward and do the improvement later.

Reply:
I've updated my suggestion. This now matches comment and non-comment values alike, while it does not touch lines, that don't start with `AllowTcpForwarding` or `#AllowTcpForwarding`.

This should make the test runs a bit more stable, as we are not relying on the exact match.

Reply:
let's keep in mind that we speaking about *test*  code not some production grade server app . So "more stable" code often translated into "test which not catching errors" . I think here exactly this case . Only hardended images suppose to have this line commented all other images suppose to have it un-commented without test code involvement